decimal64math.go

```diff
@@ -57,42 +62,70 @@ func (d Decimal64) Neg() Decimal64 {
 }
 
 // Quo computes d / e.
-func (d Decimal64) Quo(e Decimal64) Decimal64 {
-	flavor1, sign1, exp1, significand1 := d.parts()
-	flavor2, sign2, exp2, significand2 := e.parts()
-	if flavor1 == flSNaN || flavor2 == flSNaN {
-		return SNaN64
-	}
-	if flavor1 == flQNaN || flavor2 == flQNaN {
-		return QNaN64
+func (ctx Context64) Quo(d, e Decimal64) Decimal64 {
+	dp := d.getParts()
+	ep := e.getParts()
+	if ep.isNan() || dp.isNan() {
+		return *propagateNan(&dp, &ep)
 	}
-	sign := sign1 ^ sign2
-	if d == Zero64 || d == NegZero64 {
-		if e == Zero64 || e == NegZero64 {
+	var ans decParts
+	ans.sign = dp.sign ^ ep.sign
+	if dp.isZero() {
+		if ep.isZero() {
 			return QNaN64
 		}
-		return zeroes[sign]
+		return zeroes[ans.sign]
 	}
-	if flavor1 == flInf {
-		if flavor2 == flInf {
+	if dp.isinf() {
+		if ep.isinf() {
 			return QNaN64
 		}
-		return infinities[sign]
+		return infinities[ans.sign]
 	}
-	if flavor2 == flInf {
-		return zeroes[sign]
+	if ep.isinf() {
+		return zeroes[ans.sign]
 	}
-	if e == Zero64 || e == NegZero64 {
-		return infinities[sign1]
+	if ep.isZero() {
+		return infinities[dp.sign]
 	}
-	exp := exp1 - exp2 - 16
-	significand := umul64(10*decimal64Base, significand1).div64(significand2)
-	exp, significand.lo = renormalize(exp, significand.lo)
-	if significand.lo > maxSig || exp > expMax {
-		return infinities[sign]
+	dp.matchSignificandDigits(&ep)
+	ans.exp = dp.exp - ep.exp
+	for {
```

Comment:
So the idea is that there's no way of knowing how much precision the answer will take up in the final answer Without using binary long division. Here we are doing long division, but we keep adjusting the exponent (and the significand) by multiplying it by 10 until we get to the point where we've filled up our precision, then the remainder is rounded. Other methods of division cannot be used because truncation occurs without notice